iothub/device/src/Transport/Amqp/PendingAmqpTwinOperation.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+
+namespace Microsoft.Azure.Devices.Client.Transport.Amqp
+{
+    /// <summary>
+    /// A class for holding the task completion source of a pending operation and the date/time when that operation
+    /// was initiated. Because PubSub behavior has us send a request on one sending link and receive a response later on another
+    /// receiving link, we need a way to identify the request, when it was sent, and the task completion source to complete the
+    /// waiting user's task.
+    /// </summary>
+    internal class PendingAmqpTwinOperation
```

Comment:
As we discussed over a call, this could be made a bit more generic since AMQP and MQTT could use this class in the same way. We don't need to change it now, but please leave a comment in here somewhere to the effect of "future work: consolidate with PendingMqttTwinOperation"

Reply:
I'd be happy to do this work, FYI

Reply:
It's my fault that the MQTT implementation didn't follow the pattern set by the AMQP implementation

Reply:
You can do it if you like. No hurry on it as it shouldn't change behavior.